tutorials/retail_api_v2_set_inventory_java.md

```diff
@@ -0,0 +1,187 @@
+
+1. You can find the ```project_number``` in the Project Info card displayed on **Home/Dashboard**.
+
+1. Set the environment variable with the following command:
+    ```bash
+    export PROJECT_NUMBER=<YOUR_PROJECT_NUMBER>
+    ```
+
+### Google Cloud Retail libraries
+
+You can find the Java Google Retail library is described [here](https://googleapis.dev/java/google-cloud-retail/latest/index.html)
+
+## Clone the Retail code samples
+
+This step is required if this is the first Retail API tutorial you run.
+Otherwise, you can skip it.
+
+Clone the Git repository with all the code samples to learn the Retail features and check them in action.
+
+<!-- TODO(ianan): change the repository link -->
+1. Run the following command in the Terminal:
+    ```bash
+    git clone https://github.com/t-karasova/grs-retail.git
+    ```
+
+   The code samples for each of the Retail services are stored in different directories.
+
+1. Go to the code samples directory - our starting point to run more commands.
+    ```bash
+    cd java-retail/samples/snippets
+    ```
+   
+## Set inventory
+
+1. Set the following fields to send the `SetInventoryRequest` call:
+    - `inventory`—the inventory information to update. The fields allowed for updating are:
+        - `Product.priceInfo`
+        - `Product.availability`
+        - `Product.availableQuantity`
+        - `Product.fulfillmentInfo`
+    - `setMask`—indicates what product inventory fields to update
+    - `setTime`—the time when the request is sent. It is used to prevent out-of-order updates on inventory fields that contain the last update time recorded value.
+    - `allowMissing`—if set to true and the product is not found, the fulfillment information will be retained for up to 24 hours and processed after the product is created.
+
+1. Open the <walkthrough-editor-select-regex filePath="cloudshell_open/java-retail/samples/snippets/src/main/java/product/SetInventory.java" regex="getSetInventoryRequest">SetInventory.java</walkthrough-editor-select-regex> file and check the `SetInventoryRequest` request.
+
+2. Before you start, build the Maven project and go to the code samples directory - our starting point to run code samples:
+   ```bash
+   cd ~/java-retail | mvn clean install -DskipTests
+   cd ~/java-retail/samples/snippets  
+   ```
+
+3. Run the code sample in the Terminal to create a product in a catalog and remove it using a prepared request:
+    ```bash
+    mvn compile exec:java -Dexec.mainClass=product.SetInventory
+    ```
+
+1. Check the response in the Terminal. The product is initially created with `priceInfo` and `availability` information.
+
+Next, to ensure that the product inventory information was updated successfully, check the `getProduct()` response.
+
+## Send an out-of-order SetInventory request
```

Comment:
Check with Tatiana if this works or if it should be removed like in Add/Remove fulfillment places

Reply:
I checked it works. The only problem is that it not always work with the current timeout 30s. It works better with timeout 60s. I mentioned it in our spreadsheet.

Reply:
Test